test/parallel/test-child-process-spawn-argv0.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const cp = require('child_process');
+
+
+// This test spawns itself with an argument to indicate when it is a child to
+// easily and portably print the value of argv[0]
+if (process.argv[2] === 'child') {
+  console.log(process.argv0);
+  return;
+}
+
+const noArgv0 = cp.spawn(process.execPath,
+    ['test/parallel/test-child-process-spawn-argv0.js', 'child']);
```

Comment:
Also, you can write this more succinctly if you use spawnSync().  If you use spawn(), you should call `noArgv0.stdout.setEncoding('utf8')`.